lib/src/clojure_lsp/feature/drag.clj

```diff
@@ -366,27 +366,38 @@
   assume that any function with one of these names establishes bindings, even if
   it isn't in clojure.core?).
 
-  Otherwise, if the first child of `zloc` defines a local variable (according to
-  the clj-kondo analysis), then this also returns true. This allows library- and
-  user-defined code to declare that it establishes bindings, via `:lint-as`.
-  Originally this was the only heuristic, since it works for `let` and `for`.
-  But it fails for `bindings` and other forms where the clj-kondo analysis
-  reports that it merely references existing variables, rather than establishing
-  definitions."
+  Otherwise, if the `zloc`'s children are pairs of local variable definitions
+  and values (according to the clj-kondo analysis), then this also returns true.
+  This allows library- and user-defined code to declare that it establishes
+  bindings, via `:lint-as`. Originally this was the only heuristic, since it
+  works for `let` and `for`. But it fails for `bindings` and other forms where
+  the clj-kondo analysis reports that it merely references existing variables,
+  rather than establishing definitions."
   [vector-zloc uri {:keys [analysis]}]
   (boolean
-    (or (when-let [outer-zloc (z-left vector-zloc)]
-          (and (z-leftmost? outer-zloc)
-               (contains? common-binding-syms (z/sexpr outer-zloc))))
-        (let [z-pos   (z-cursor-position (z-down vector-zloc))
-              z-scope {:name-row     (:row z-pos)
-                       :name-col     (:col z-pos)
-                       :name-end-row (:end-row z-pos)
-                       :name-end-col (:end-col z-pos)}]
-          (q/find-first (fn [element]
-                          (and (= :locals (:bucket element))
-                               (shared/inside? element z-scope)))
-                        (get analysis (shared/uri->filename uri)))))))
+    (or (when-let [op-zloc (z-left vector-zloc)]
+          (and (z-leftmost? op-zloc)
+               (contains? common-binding-syms (z/sexpr op-zloc))))
+        (let [child-nodes (filter n/sexpr-able? (n/children (z/node vector-zloc)))
+              node-scope (fn [node]
+                           (let [nmeta (meta node)]
+                             {:name-row     (:row nmeta)
+                              :name-col     (:col nmeta)
+                              :name-end-row (:end-row nmeta)
+                              :name-end-col (:end-col nmeta)}))]
+          (when (even? (count child-nodes))
+            (let [analysis (filter #(shared/inside? % (node-scope (z/node vector-zloc)))
+                                   (get analysis (shared/uri->filename uri)))]
+              (->> child-nodes
+                   (partition 2)
+                   (every? (fn [[lvar rvar]]
```

Comment:
Sometimes I forgot how partition could be so handful

Reply:
👍 it's a useful tool. I recently re-discovered `list*`, which is useful for constructing s-expressions.